src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.cs

```diff
@@ -571,17 +597,32 @@ public Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, HttpCompl
                 if (NetEventSource.IsEnabled) NetEventSource.ClientSendCompleted(this, response, request);
                 return response;
             }
+            catch (TaskCanceledException e) when (TimeoutFired(timeoutState, callerToken))
+            {
+                ThrowTimeoutException(e);
+                throw;
+            }
             catch (Exception e)
             {
                 HandleFinishSendAsyncError(e, cts);
                 throw;
             }
             finally
             {
-                HandleFinishSendAsyncCleanup(cts, disposeCts);
+                HandleFinishSendAsyncCleanup(cts, disposeCts, timer);
             }
         }
 
+        private bool TimeoutFired(TimeoutState timeoutState, CancellationToken callerToken)
+            => !callerToken.IsCancellationRequested && !_pendingRequestsCts.IsCancellationRequested && timeoutState != null && timeoutState.IsFired;
```

Comment:
_pendingRequestsCts gets assigned to a new instance as part of CancelPendingRequests before the instance that was there is even canceled.  I don't think this check on it is meaningful.

Reply:
You are right, it gets set to a new instance. Check removed.